pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -489,9 +492,39 @@ func (c *Compactor) runCompact(ctx context.Context, logger log.Logger, job Job,
 			return level.Error(logger).Log("msg", "failed to compact new chunks", "err", err)
 		}
 
+		archivePath := filepath.Join(c.cfg.WorkingDirectory, storedBlock.BlockPath[strings.LastIndex(storedBlock.BlockPath, "/")+1:])
+		archiveFile, err := os.Create(archivePath)
+		if err != nil {
+			return err
+		}
+		defer archiveFile.Close()
+		defer func() {
+			err = os.Remove(archivePath)
+			if err != nil {
+				level.Error(logger).Log("msg", "removing archive file", "err", err, "file", archivePath)
+			}
+		}()
```

Comment:
This may lead to a problem: deferred function are executed in the reverse order of their deferring.

Reply:
TIL.  Thank you